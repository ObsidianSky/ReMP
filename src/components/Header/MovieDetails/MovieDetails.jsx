import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './MovieDetails.scss'

export const MovieDetails = ({
    img,
    title,
    year,
    genre,
    rating,
    duration,
    description,
    director,
    cast
}) => (
    <div className="movie-details">
        <div className="movie-details__poster">
            <img src={img} className="movie-details__img"/>
        </div>
        <div className="movie-details__info">
            <div className="movie-details__heading">
                <div className="movie-details__title">{title}</div>
                <div className="movie-details__rating">{rating}</div>
            </div>
            <div className="movie-details__genre">{genre}</div>
            <div className="movie-details__meta">
                <div className="movie-details__year">{year}</div>
                <div className="movie-details__duration">{duration}</div>
            </div>
            <div className="movie-details__description">{description}</div>
            { director && <div className="movie-details__director">Director: { director.name}</div> }
            <div className="movie-details__cast">Cast: {cast}</div>
        </div>
    </div>
);

MovieDetails.propTypes = {
    img: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    rating: PropTypes.string.isRequired,
    duration: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.shape({
        name: PropTypes.string.isRequired
    }),
    cast: PropTypes.string.isRequired
};

const mapStateToProps = ({ movies }) => ({ ...movies.selectedMovie });

export default connect(mapStateToProps)(MovieDetails);