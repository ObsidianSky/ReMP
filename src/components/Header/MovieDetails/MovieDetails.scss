@import '../../../assets/styles/variables';

.movie-details {
  display: flex;

  &__poster {
    margin-right: 5%;
    flex: 1 0 30%;
  }

  &__info {
    flex: 1 0 65%;
  }

  &__heading {
    display: flex;
    align-items: center;
    margin-bottom: 4px;
  }

  &__title {
    font-size: 32px;
    font-weight: bold;
    color: $pink;
    margin-right: 20px;
  }

  &__rating {
    color: $green;
    font-size: 22px;
    width: 40px;
    height: 40px;
    line-height: 40px;
    text-align: center;
    border-radius: 50%;
    border: 1px solid $green;
  }

  &__genre {
    font-size: 15px;
    color: $text;
    margin-bottom: 15px;
  }

  &__meta {
    font-size: 15px;
    color: $text-light;
    margin-bottom: 30px;
    display: flex;
  }

  &__year {
    margin-right: 20px;
  }

  &__description {
    font-size: 17px;
    color: $text;
    margin-bottom: 20px;
  }

  &__director {
    font-size: 14px;
    color: $text-dark;
    margin-bottom: 10px;
  }

  &__cast {
    font-size: 14px;
    color: $text-dark;
  }
}