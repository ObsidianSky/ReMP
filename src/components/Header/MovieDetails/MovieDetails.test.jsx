import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MovieDetails } from './MovieDetails';

describe('MovieDetails', () => {
    let props;

    beforeEach(() => {
        props = {
            img: 'img',
            title: 'title',
            year: 'year',
            genre: 'genre',
            rating: 'rating',
            duration: 'duration',
            description: 'description',
            cast: 'cast'
        };
    });

    describe('when we do NOT have director', () => {
        test('should renders correctly', () => {
            const wrapper = shallow(<MovieDetails { ...props }/>);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when we have director', () => {
        beforeEach(() => {
            props.director = {
                name: 'director'
            };
        });
        test('should renders correctly', () => {
            const wrapper = shallow(<MovieDetails { ...props }/>);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});