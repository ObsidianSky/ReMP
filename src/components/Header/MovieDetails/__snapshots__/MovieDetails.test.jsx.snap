// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails when we do NOT have director should renders correctly 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details__poster"
  >
    <img
      className="movie-details__img"
      src="img"
    />
  </div>
  <div
    className="movie-details__info"
  >
    <div
      className="movie-details__heading"
    >
      <div
        className="movie-details__title"
      >
        title
      </div>
      <div
        className="movie-details__rating"
      >
        rating
      </div>
    </div>
    <div
      className="movie-details__genre"
    >
      genre
    </div>
    <div
      className="movie-details__meta"
    >
      <div
        className="movie-details__year"
      >
        year
      </div>
      <div
        className="movie-details__duration"
      >
        duration
      </div>
    </div>
    <div
      className="movie-details__description"
    >
      description
    </div>
    <div
      className="movie-details__cast"
    >
      Cast: 
      cast
    </div>
  </div>
</div>
`;

exports[`MovieDetails when we have director should renders correctly 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details__poster"
  >
    <img
      className="movie-details__img"
      src="img"
    />
  </div>
  <div
    className="movie-details__info"
  >
    <div
      className="movie-details__heading"
    >
      <div
        className="movie-details__title"
      >
        title
      </div>
      <div
        className="movie-details__rating"
      >
        rating
      </div>
    </div>
    <div
      className="movie-details__genre"
    >
      genre
    </div>
    <div
      className="movie-details__meta"
    >
      <div
        className="movie-details__year"
      >
        year
      </div>
      <div
        className="movie-details__duration"
      >
        duration
      </div>
    </div>
    <div
      className="movie-details__description"
    >
      description
    </div>
    <div
      className="movie-details__director"
    >
      Director: 
      director
    </div>
    <div
      className="movie-details__cast"
    >
      Cast: 
      cast
    </div>
  </div>
</div>
`;
